lib/core/elastic_connector_actions.rb

```diff
@@ -114,11 +114,24 @@ def claim_job(connector_id)
           :connector_id => connector_id,
           :status => Connectors::SyncStatus::IN_PROGRESS,
           :worker_hostname => Socket.gethostname,
-          :created_at => Time.now
+          :created_at => Time.now,
+          :filtering => convert_connector_filtering_to_job_filtering(connector_record.dig('_source', 'filtering'))
         }
-        job = client.index(:index => Utility::Constants::JOB_INDEX, :body => body)
 
-        job['_id']
+        client.index(:index => Utility::Constants::JOB_INDEX, :body => body)
+      end
+
+      def convert_connector_filtering_to_job_filtering(connector_filtering)
```

Comment:
Nit: this should not be a public method - maybe a can be moved to Utility namespace?
Later we can provide different abstraction over it, I feel like `ElasticConnectorActions` becomes too complex - initial idea over it was to provide a domain interface over Elasticsearch, so just lightweight wrapper over the index. I'll think a bit about how to make it simpler.

Reply:
I had the same thought. However, I like being able to unit test just this bit of logic on its own, without having to mock Elasticsearch queries and responses. I've been told by mentors before that it's an anti-pattern to make a method private and then to write unit tests for it where you exercise access to a method that you shouldn't have direct access to. I know I _can_ do this with `send()`, but I'm hesitant to. WDYT?

Reply:
Makes sense to me, thanks for the explanation :)